### PR TITLE
Bc dspdc 463 create object

### PR DESCRIPTION
**Pull request by benjamincarlin**

### CODE REVIEW

File: gcs/src/main/scala/org/broadinstitute/monster/storage/gcs/GcsApi.scala [hunk 1]
```diff
@@ -168,7 +168,42 @@ class GcsApi private[gcs] (runHttp: Request[IO] => Resource[IO, Response[IO]]) {
...
-  // TODO: if the data is greater than n bytes use initResumableUpload and the uploadBytes in multiple chunks
+  def createObject(
+    bucket: String,
+    path: String,
+    contentType: `Content-Type`,
+    expectedSize: Long,
+    expectedMd5: Option[String],
+    data: Stream[IO, Byte]
+  ): IO[Unit] = {
+    if (expectedSize < MaxBytesPerUploadRequest.toLong) {
```
Comment (danxmoran) [inline gcs/src/main/scala/org/broadinstitute/monster/storage/gcs/GcsApi.scala]: Do we ever need `MaxBytesPerUploadRequest` as an int? If not, we can type it as a `Long` at its definition site and avoid running this conversion every time.

File: gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala [hunk 2]
```diff
@@ -37,6 +37,86 @@ class GcsApiSpec extends FlatSpec with Matchers with OptionValues with EitherVal
...
+    )
+
+    io.circe.parser
+      .parse(new String(chunk.toArray[Byte]))
+      .right
+      .value shouldBe buildUploadMetadata(path, expectedMd5)
+    ()
+  }
+
+  private def assertInitResumableUploadResponse(
```
Comment (danxmoran) [inline gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala]: I don't see much value in separating the assertions from the response generation (if we get an upload initialization request, we always want to run those assertions and then return a matching response). I'd merge this method with the one above.

Same for the other request/response pairs.

File: gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala [hunk 3]
```diff
@@ -30,13 +29,19 @@ class GcsApiIntegrationSpec
...
+      .flatMap(s => Stream.emits(s.getBytes))
+      .take(n.toLong)
+  }
 
-  private val bodyMd5 = DigestUtils.md5Hex(bodyText)
+  // smaller than MaxBytesPerUploadRequest
+  private val bodySize = 2 * GcsApi.ChunkSize
+  private val body = bodyText(2 * GcsApi.ChunkSize)
+  // TODO: best way to get Md5
+  private val bodyMd5 = DigestUtils.md5Hex(buildString(body).unsafeRunSync().toString())
```
Comment (danxmoran) [inline gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala]: I think the top-level `body` and `bodyMd5` should go away, so tests are always forced to be explicit about how they're generating the tested text.

File: gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala [hunk 4]
```diff
@@ -79,8 +84,9 @@ class GcsApiIntegrationSpec
     val blob = BlobId.of(bucket, blobPath)
     val blobInfo = BlobInfo.newBuilder(blob).setContentType("text/plain").build()
 
+    // TODO best way to get bytes?
```
Comment (danxmoran) [inline gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala]: I would have `writeTestFile` (and `writeGzippedTestFile`) take a `Chunk[Byte]` as an argument. Then `setup`can use that chunk.

File: gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala [hunk 5]
```diff
@@ -122,7 +126,7 @@ class GcsApiIntegrationSpec
   it should "read entire objects as a stream" in {
     writeTestFile.use { blob =>
       withClient(api => buildString(api.readObject(blob.getBucket, blob.getName)))
-    }.unsafeRunSync() shouldBe bodyText
+    }.unsafeRunSync() shouldBe buildString(body).unsafeRunSync()
```
Comment (danxmoran) [inline gcs/src/it/scala/org/broadinstitute/monster/storage/gcs/GcsApiIntegrationSpec.scala]: It'd be better to only `unsafeRunSync()` once per test. In this case, the setup would be like:
```scala
val checks = for {
  chunk <- bodyText(someNumber).compile.toChunk
  writtenBytes <- writeTestFile(chunk).use { blob =>
    withClient(...)
  }
  _ <- writtenBytes shouldBe chunk
} yield ()
checks.unsafeRunSync()
```

**Review by danxmoran — COMMENTED** [review]
Some initial comments to get started. Still looking through the integration tests.

**Review by raaidbroad — APPROVED** [review]

**Review by danxmoran — APPROVED** [review]

**Pull request merged**